app/view/combo/Gazetteer.js

```diff
@@ -46,17 +46,35 @@ Ext.define('CpsiMapview.view.combo.Gazetteer', {
             })
         });
 
-        me.on('beforequery', function () {
-            var val = this.getRawValue();
-            var url = me.url + encodeURIComponent(val.trim());
-            me.getStore().getProxy().setUrl(url);
-        });
+        me.callParent();
 
-        me.on('afterrender', function () {
-            me.locationLayer.set('displayInLayerSwitcher', false);
+        me.on({
+            beforequery: function () {
+                var val = this.getRawValue();
+                // set a key on the layer using the unique id of the combo
+                me.locationLayer.set('layerKey', me.id);
+                if (Ext.isEmpty(BasiGX.util.Layer.getLayersBy('layerKey', me.id))) {
+                    // if the layer is not yet in the map then add it
+                    if (me.map) {
+                        me.map.addLayer(me.locationLayer);
```

Comment:
Are there any conflicts with the adding the layer adding to the map by the base class?

Reply:
Oh, OK. Then it is good to go. Thanks for clarification.